Please find below the Project for Statistical Learning course. This is an individual assignment. Kindly submit it before it's deadline.

The Titan Insurance Company has just installed a new incentive payment scheme for its lift policy sales force. It wants to have an early view of the success or failure of the new scheme. Indications are that the sales force is selling more policies, but sales always vary in an unpredictable pattern from month to month and it is not clear that the scheme has made a significant difference.

Life Insurance companies typically measure the monthly output of a salesperson as the total sum assured for the policies sold by that person during the month. For example, suppose salesperson X has, in the month, sold seven policies for which the sums assured are £1000, £2500, £3000, £5000, £10000, £35000. X's output for the month is the total of these sums assured, £61,500. Titan's new scheme is that the sales force receives low regular salaries but are paid large bonuses related to their output (i.e. to the total sum assured of policies sold by them). The scheme is expensive for the company, but they are looking for sales increases which more than compensate. The agreement with the sales force is that if the scheme does not at least break even for the company, it will be abandoned after six months.

The scheme has now been in operation for four months. It has settled down after fluctuations in the first two months due to the changeover.

To test the effectiveness of the scheme, Titan have taken a random sample of 30 salespeople measured their output in the penultimate month prior to changeover and then measured it in the fourth month after the changeover (they have deliberately chosen months not too close to the changeover). The outputs of the salespeople are shown in Table 1

 

SALESPERSON

Old Scheme (in thousands)

New Scheme (in thousands)

Questions
Find the mean of old scheme and new scheme column. (5 points)
Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? (10 points)
What conclusion does the test (p-value) lead to? (2.5 points)
Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:
        a) The probability of a type 1 error? (2.5 points)

        b) What is the p- value of the hypothesis test if we test for a difference of $5000? (5 points)

        c) Power of the test (5 points)

In [85]:
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats
import math

In [83]:
scheme_data=pd.read_csv("data.csv", index_col=0)
#renaming the columns
scheme_data.columns=['Old_Scheme','New_Scheme']
scheme_data.head()

Old_Scheme  New_Scheme
SALESPERSON                        
1                    57          62
2                   103         122
3                    59          54
4                    75          82
5                    84          84

In [57]:
describe_data=scheme_data.describe(include="all")
describe_data

Old_Scheme  New_Scheme
count   30.000000   30.000000
mean    68.033333   72.033333
std     20.455980   24.062395
min     28.000000   32.000000
25%     54.000000   55.000000
50%     67.000000   74.000000
75%     81.500000   85.750000
max    110.000000  122.000000

In [67]:
#Find the mean of old scheme and new scheme column.
#===============================================
describe_data.loc['mean']

Old_Scheme    68.033333
New_Scheme    72.033333
Name: mean, dtype: float64

In [ ]:
#level of significance is 0.05
#The sample size here is 30 , i.e. N = 30
# Null Hypothesis - there is no significant raise between output of oldscheme and newscheme i.e. Mew_old- Mew_new=0
# Alternate Hypothesis - there is a significant raise between the outpu of old scheme and newscheme i.e. Mew_old- Mew_new >0

#As the data is a paired observation we will use paired ttest

In [69]:
#Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? 
#=====================================================================
#checking normality using shapiro test and checking variance using levene test
stats.shapiro(scheme_data.Old_Scheme),stats.shapiro(scheme_data.New_Scheme),\
stats.levene(scheme_data.Old_Scheme, scheme_data.New_Scheme)

#data follow normality as pvalue for both coulmns is  P>0.05
#p value p>0.05 in levene test hence all variances are equal


((0.9885101914405823, 0.9813658595085144),
 (0.9687567353248596, 0.5057420134544373),
 LeveneResult(statistic=1.063061539437244, pvalue=0.30679836081811235))

In [106]:
#Applying paired t-test
t, p = stats.ttest_rel(scheme_data.Old_Scheme, scheme_data.New_Scheme)
print ('t value:', t)

#The p value which we got above is for two tailed test, in our case we need for single tailed hence
p_value=1-stats.t.cdf(1.5559,30)
print ('P value:', p_value, '\n')
#P value is 0.065>0.05

t value: -1.5559143823544377
P value: 0.06510979171859932 



What conclusion does the test (p-value) lead to? (2.5 points)
.====================================================
"If P is low H0 should go."
On the contrary here P-value is 0.0651 which is greater than 0.05  which means that we cannot reject null hypothesis  i.e. 
there is no significant increase in the output of old scheme and newscheme.
Conclusion: The new scheme did not raise the output significantly. 

In [102]:
#Here given that the average output must increase by £5000 . So,
#Null Hypothesis = Mew_old- Mew_new=0
#Alternate hypothesis Mew_old- Mew_new = £5000

#a) The probability of a type 1 error?
#==========================================

#probability of type1 error is the given significance level alpha=0.05. So there is 5% probability we make Type I error


In [103]:
#b) What is the p- value of the hypothesis test if we test for a difference of $5000?
#====================================================================================
#degree freedom is 29 as sample size is 30


t_stat=abs(stats.t.ppf(0.05,df=29))
t_stat

1.6991270265334977

In [112]:
## getting Difference between values of  old scheme and ne scheme 
scheme_data['Difference'] =  scheme_data['New_Scheme'] -scheme_data['Old_Scheme'] 
scheme_data

Old_Scheme  New_Scheme  Difference
SALESPERSON                                    
1                    57          62           5
2                   103         122          19
3                    59          54          -5
4                    75          82           7
5                    84          84           0
6                    73          86          13
7                    35          32          -3
8                   110         104          -6
9                    44          38          -6
10                   82         107          25
11                   67          84          17
12                   64          85          21
13                   78          99          21
14                   53          39         -14
15                   41          34          -7
16                   39          58          19
17                   80          73          -7
18                   87          53         -34
19                   73          66          -7
20                   65          78          13
21                   28          41          13
22                   62          71           9
23                   49          38         -11
24                   84          95          11
25                   63          81          18
26                   77          58         -19
27                   67          75           8
28                  101          94          -7
29                   91         100           9
30                   50          68          18

In [76]:
std_of_diff=scheme_data['Difference'].std()
std_of_diff

14.08104619937615

In [86]:
#we need to find the D_bar
#square root of 30 = 5.48
#applying paired t_test calculation 
D_bar=(t_stat*std_of_diff)/math.sqrt(30)
D_bar

4.36817615622872

In [98]:
#now given that Mew_old- Mew_new = £5000
t_stat_5k=((D_bar-5)*math.sqrt(30))/std_of_diff
t_stat_5k

-0.24576595140954954

In [100]:
#so,the p- value of the hypothesis test if we test for a difference of $5000 is beta=40.3%
p_value_5k=stats.t.cdf(t_stat_5k,df=29)
p_value_5k

0.4037972524794317

In [113]:
#c) Power of the test (5 points)
#===============================

import statsmodels.stats.power as smp
meanDiff =scheme_data['Difference'].mean()
meanStd = scheme_data['Difference'].std()
effect_size = meanDiff/meanStd
nobs = 30
alpha = 0.05
smp.ttest_power(effect_size, nobs, alpha, alternative='larger')

0.4501248712978272